test/unit/query/builder.js

```diff
@@ -2,21 +2,31 @@
 /*eslint no-var:0, indent:0, max-len:0 */
 'use strict';
 
-var MySQL_Client = require('../../../lib/dialects/mysql')
-var PG_Client = require('../../../lib/dialects/postgres')
-var Oracle_Client = require('../../../lib/dialects/oracle')
-var Oracledb_Client = require('../../../lib/dialects/oracledb')
-var SQLite3_Client = require('../../../lib/dialects/sqlite3')
-var MSSQL_Client = require('../../../lib/dialects/mssql')
+var _typeof2 = require('babel-runtime/helpers/typeof');
+
+var _typeof3 = _interopRequireDefault(_typeof2);
+
+var _keys = require('babel-runtime/core-js/object/keys');
```

Comment:
you could try if it works that way, I think that the whole PR will be a lot less intrusive and I would feel better about it. 

If that doesn't work either, I think that I have to look through query builder code and see if it is easy enough to fix that clients of subqueries are automatically patched to use client of the base query, fix that and then get back to this PR.